core/src/main/java/me/leoko/advancedban/manager/PunishmentManager.java

```diff
@@ -126,12 +127,17 @@ public void discard(String name) {
      * @return the punishments
      */
     public List<Punishment> getPunishments(String target, PunishmentType put, boolean current) {
```

Comment:
The `#getPunishments` method is part of the general API. This method may be used in other places than just by the `/history` command. So this logic should be extracted to the history command.

One idea would be to add another `getPunishments()` which accepts a list of PunishmentTypes then the history command could use that method